### DISCUSSION TITLE
cron job reconnect PPPoE

### DISCUSSION THREAD

**Discussion by a-kempka** [unifi-utilities/unifi-common#126] [Q&A] [answered]
How do cron jobs work that are executed at a specific time and not after booting up the UDMP?

I'd like to set up a PPPoE reconnect at 3am that persists thorughout reboots and firmware updates:
`0   3   *   *   *   root   /usr/bin/killall -HUP pppd`

Would putting the above command it in /mnt/data/on_boot.sh/reconnect.sh (in an executable script with `#!/bin/sh`, of course) spawn a new cron job after each reboot so that it gets dublicated every time the UDMP reboots?
What are the rules for the numbered prefixes in `/mnt/data/on_boot.d/`? (would this mean the file should be named `10-reconnect.sh`?

**Comment by ilkertanli** [comment, marked as answer]
Save below script under /mnt/data/on_boot.d/ and change the reboot command with whatever you want to do (killall -HUP pppd). Script checks the date when it matches to a specified time 05:00:00 it executes. 


#!/bin/sh
while true;
do
    DATE=`date | cut -d' ' -f4`
    if [[ $DATE == "05:00:00" ]]
    then
            reboot
            sleep 30s
    fi
done
